src/main/java/com/mineinabyss/looty/ecs/action/DecreaseDurabilityAction.kt

```diff
@@ -0,0 +1,17 @@
+package com.mineinabyss.looty.ecs.action
+
+import com.mineinabyss.geary.ecs.api.actions.GearyAction
+import com.mineinabyss.geary.ecs.api.entities.GearyEntity
+import com.mineinabyss.looty.ecs.components.DurabilityComponent
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+@Serializable
+@SerialName("looty:decreaseDurability")
+class DecreaseDurabilityAction(private val deltaDurability: Int) : GearyAction() {
+    override fun GearyEntity.run(): Boolean {
+        val durability = get<DurabilityComponent>() ?: return false
```

Comment:
By the way, I think at the top of the class you can use 
```kotlin
val GearyEntity.location by get<DurabilityComponent>()
```
Just like you would with conditions. I'm working on unifying this with the query system but the syntax shouldnt change much.